### PR TITLE
feat(messenger): add parser of Nickname packet

### PR DESCRIPTION
**Pull request by NamsooCho**

### CODE REVIEW

File: src/toxcore/messenger/packet/nickname.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+/** Nickname is a struct that holds string of my nickname.
+
+This packet is used to transmit sender's nickname to a friend.
+Every time a friend become online or my nickname is changed,
+this packet is sent to the friend or to all friends of mine.
+
+*/
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Nickname {
+    nickname: Vec<u8>,
```
Comment (kpp) [inline src/toxcore/messenger/packet/nickname.rs]: Please make a research which encoding could be used in nickname.

File: src/toxcore/messenger/packet/nickname.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+impl FromBytes for Nickname {
+    named!(from_bytes<Nickname>, do_parse!(
+        tag!("\x30") >>
+        nickname: verify!(rest, |nickname: &[u8]| nickname.len() <= MAX_NICKNAME_DATA_SIZE) >>
```
Comment (kpp) [inline src/toxcore/messenger/packet/nickname.rs]: How about termination null?

File: src/toxcore/messenger/packet/nickname.rs (hunk 1 above)
Comment (kpp) [reply src/toxcore/messenger/packet/nickname.rs]: Then we should check that the byte sequence is a correct UTF-8 string and cast `Vec<u8>` to String.

Comment (NamsooCho) [reply src/toxcore/messenger/packet/nickname.rs]: done

File: src/toxcore/messenger/packet/nickname.rs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    // Test for encoding error of from_bytes
+    #[test]
+    fn nickname_from_bytes_encoding_error() {
+        let err_string = vec![0, 159, 146, 150]; // not UTF8 bytes
+        assert!(Nickname::from_bytes(&err_string).is_err());
+    }
+
+    // Test for encoding error of to_bytes
+    #[test]
+    fn nickname_to_bytes_encoding_error() {
```
Comment (kpp) [inline src/toxcore/messenger/packet/nickname.rs]: Split into 2 tests: 1 for overflow and 1 for bad encoding

File: src/toxcore/messenger/packet/nickname.rs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+pub struct Nickname {
+    nickname: Vec<u8>,
+}
+
+impl FromBytes for Nickname {
+    named!(from_bytes<Nickname>, do_parse!(
+        tag!("\x30") >>
+        nickname: verify!(rest, |nickname: &[u8]| nickname.len() <= MAX_NICKNAME_DATA_SIZE
+            && str::from_utf8(nickname).is_ok()) >>
+        (Nickname { nickname: nickname.to_vec() })
```
[the remark this answers is not in the document]
Comment (NamsooCho) [reply src/toxcore/messenger/packet/nickname.rs]: done

File: src/toxcore/messenger/packet/nickname.rs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Nickname {
+    nickname: String,
+}
+
+impl FromBytes for Nickname {
+    named!(from_bytes<Nickname>, do_parse!(
+        tag!("\x30") >>
+        nickname: map_res!(rest, str::from_utf8) >>
+        verify!(value!(nickname), |nickname: &str| nickname.len() <= MAX_NICKNAME_DATA_SIZE) >>
```
Comment (kurnevsky) [inline src/toxcore/messenger/packet/nickname.rs]: I'd check the length first, only after convert to string, like:
```rust
map_res!(
    verify!(rest, |nickname: &[u8]| nickname.len() <= MAX_NICKNAME_DATA_SIZE),
    str::from_utf8
)
```

Comment (NamsooCho) [reply src/toxcore/messenger/packet/nickname.rs]: done

**Pull request merged**